src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1502,6 +1502,34 @@ void QgsOgrProvider::loadMetadata()
         }
       }
     }
+    else if ( ( mGDALDriverName == QLatin1String( "ESRI Shapefile" ) ) )
+    {
+      // look for .shp.xml sidecar file
+      const QString sidecarPath = mFilePath + ".xml";
```

Comment:
Not a blocker for this PR, just putting it out there: wondering if it'd eventually be worth to handle zipped shapefiles here.

Reply:
Good point... non trivial though, since we'd need to extract the file and then read. I'll defer that till I have a need :)

Reply:
Yeah -- we also fail to handle other sidecar files (e.g. .qml style) so it's a built-in expectation of failure ;)